### ISSUE TITLE
Uncaught TypeError, fresh install, parse 1.6.9

### ISSUE THREAD

**Issue by joaomarceloods** [brandid/parse-angular-patch#19]
Has anyone gone through this? Is there a fix?

I just installed _parse-angular-patch_ from bower. I get the following error in the browser console as soon is the page is loaded:

```
Uncaught TypeError: Cannot read property 'prototype' of undefined`
    (anonymous function)    @   parse-angular.js:66
    (anonymous function)    @   parse-angular.js:64
```

I believe that _'prototype' of undefined_ refers to `Parse[currentClass].prototype[method]`:

```
currentProtoMethods.forEach(function(method) { << LINE 64

    var origMethod = Parse[currentClass].prototype[method]; << LINE 66

    // Overwrite original function by wrapping it with $q
    Parse[currentClass].prototype[method] = function() {

        return origMethod.apply(this, arguments)
            .then(function(data) {
                var defer = $q.defer();
                defer.resolve(data);
                return defer.promise;
            }, function(err) {
                var defer = $q.defer();
                defer.reject(err);
                return defer.promise;
            });

    };

});
```

I used _src_ version of parse-angular-patch to get the line number for the error, but it happens on _dist_ version too. This is my configuration... This is also the order that I am importing the scripts in HTML:

```
parse 1.6.9
angular 1.4.8
parse-angular-patch 0.1.2
```

**Comment by yundom** [comment]
A temporary solution is just comment out the `Collection` element of the `methodsToUpdate` array.

```
                var methodsToUpdate = {
                    "Object": {
                        prototype: ['save', 'fetch', 'destroy'],
                        static: ['saveAll', 'destroyAll']
                    },
                    // "Collection": {
                    //  prototype: ['fetch'],
                    //  static: []
                    // },
                    "Query": {
                        prototype: ['find', 'first', 'count', 'get'],
                        static: []
                    },
                    "Cloud": {
                        prototype: [],
                        static: ['run']
                    },
                    "User": {
                        prototype: ['signUp'],
                        static: ['requestPasswordReset', 'logIn']
                    },
                    "FacebookUtils": {
                        prototype: [],
                        static: ['logIn', 'link', 'unlink']
                    },
                    "Config": {
                        prototype: [],
                        static: ['get']
                    }
                };
```

**Comment by AhmedBytesBits** [comment]
I am getting same issue here

Angular: 1.4.8
Parse:1.6.12
Parse-angular-patch: 0.1.2

Solution suggested yundom works but not sure of side-effects

**Comment by cleever** [comment]
Same here

**Comment by milen-3web** [comment]
Same here
error    Uncaught TypeError: Cannot read property 'prototype' of undefined, /lib/parse-angular-patch/dist/parse-angular.js, Line: 1

**Comment by peturh** [comment]
My code uses `Parse.Collection.extend`, what am I supposed to do?

**Comment by qorpus** [comment]
Also getting the same issue

Angular - 1.5.5
Parse - 1.6.14
Parse-angular-path: 0.1.2

**Comment by peturh** [comment]
I forked it, I change my code to not use collection and basically commented it out in this fork: https://github.com/peturh/parse-angular-patch

Built it with gulp so maybe you can just copy the `dist/` file.

**Comment by qorpus** [comment]
yeah - the key in the change log Collections no longer exists in Parse 

> v1.6.0 — September 9, 2015
> 
> New Features
> 
> Designed to work with ES6. Parse.Object can be treated as a class, and can be subclassed with the >extend keyword
> 
> When subclassing Parse.Object, you will need to call Parse.Object.registerSubclass()
> 
> Better support for different platforms, with custom builds for React Native and Node
> 
> All platforms still use the “parse” npm module, but node.js applications should use >“require(‘parse/node’)” and React Native applications should use “require(‘parse/react-native’)”
> 
> Objects can be created from JSON with Parse.Object.fromJSON()
> 
> Pointers can be generated with Parse.Object.prototype.toPointer()
> 
> All SDK configuration is now stored in Parse.CoreManager
> 
> Controllers for concepts like local storage can be hot-swapped via CoreManager
> 
> Changes
> 
> SDK no longer contains Backbone-specific behavior. Moving forward, the core SDK will not be tied to >any single framework, but we will work with the community to produce up-to-date bindings like >Parse+React. The major changes are the removal of Parse.Collection, and allowing Parse.Objects to >act as event channels.
> 
> SDK no longer ships with Underscore.js included.
> 
> Field type mismatches (trying to increment a string field) will now throw exceptions on the client side.
> 
> In server environments where the current user has not been explicitly enabled, fetching the current >user will now return null instead of throwing. Attempts to change the current user through logIn or >become will still throw...
> 
> Parse.Ops no longer publicly expose their values

Thanks @peturh - ill try it out

**Comment by qorpus** [comment]
Also Underscore is removed, so unless you include underscore yourself this will still fail :\

**Comment by peturh** [comment]
Well, my small edit works for me. Please contribute if you can find anything better.
